tests/test_helper.py

```diff
@@ -17,3 +20,119 @@
     FONT = (
         "Liberation-Mono"  # This is available in the fonts-liberation package on Linux.
     )
+
+
+def bitmap_to_frame_list(bitmap_frames, color_dict=None):
+    """
+    Turns a list of strings into a list of numpy arrays, each letter representing a (r, g, b) tuple.
+    Example input (2 frames, 5x3 pixel size):
+    [["RRRRR",
+      "RRBRR",
+      "RRBRR"],
+     ["RGGGR",
+      "RGGGR",
+      "RGGGR"]]
+    """
+    # "O" represents black
+    # "X", "Y", "Z" are arbitrary colours
+    if color_dict is None:
+        color_dict = {
+            "R": (255, 0, 0),
+            "G": (0, 255, 0),
+            "B": (0, 0, 255),
+            "O": (0, 0, 0),
+            "W": (255, 255, 255),
+            "X": (89, 225, 62),
+            "Y": (113, 157, 108),
+            "Z": (215, 182, 143),
+        }
+
+    output_array = []
+    for input_frame in bitmap_frames:
+        output_frame = []
+        for row in input_frame:
+            output_frame.append([color_dict[color] for color in row])
+        output_array.append(np.array(output_frame))
+
+    return output_array
+
+
+def bitmap_to_clip(bitmap_frames, color_dict=None):
+    """
+    Takes input of the same form as `bitmap_to_frame_list`, but then creates a VideoClip with the
+    result. The VideoClip has length in seconds equal to the number of frames, and fps is 1.
+    :param bitmap_frames:
+    :return:
+    """
+    frame_list = bitmap_to_frame_list(bitmap_frames, color_dict=color_dict)
+    make_frame = lambda t: frame_list[t]
+    return VideoClip(make_frame=make_frame, duration=len(frame_list) - 1).set_fps(1)
+
+
+def clip_to_frame_list(clip):
+    """
+    Creates a list containing each frame in `clip` as a numpy array.
+    """
+    li = []
+    for t in range(clip.duration * clip.fps):
+        frame = clip.get_frame(t)
+        li.append(frame)
+    return li
+
+
+def clip_frames_equal(clip1, clip2):
```

Comment:
two observations here. Firstly, I guess this is basically what numpy's assertions does

 https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_equal.html
and 
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose

Anyway, an interesting approach would be to define `__eq__` in the Clip class, making this frame by frame comparison, and implement the verbose output at pytest level 
https://docs.pytest.org/en/latest/assert.html#defining-your-own-explanation-for-failed-assertions